### ISSUE TITLE
Cannot get builders from registry for ParticleEffect sub types

### ISSUE THREAD

**Issue by Deamon5550** [SpongePowered/SpongeAPI#926]
Currently you are unable to get instances of the builders for any of the particle subtypes (Colored etc.). This is due to the fact that the createBuilder method in the GameRegistry is looking for a type of `<T extends ResettableBuilder<T>>` and the builders for the particle subtypes extend the common ParticleEffect builder which extends `ResettableBuilder<ParticleEffect.Builder>`, whereas the subtype builders need to be unltimately extending `ResettableBuilder<SubTypeParticle.Builder>`.

We might be able to add a generic argument to ParticleEffect.Builder to pass to the ResettableBuilder but I'd need to look into it a bit more.

**Comment by gratimax** [comment]
I thought I fixed this in 92a9d1bd595939a. The signature of the createBuilder method is now:

```java
<T extends ResettableBuilder<? super T>> T createBuilder(Class<T> builderClass)
```

**Comment by Deamon5550** [comment]
Oh you are correct <.<

Apparently when you open a class from the repository search page it opens it up referencing the tree of when that file was last modified (which for ColoredParticle was 2a7c654 6 days ago) so when I navigated to the GameRegistry from there I was looking at a 6 day old game registry.

My bad. I guess I should use the Find File feature rather than the search box, it doesn't seem to have this issue.
